nevergrad/benchmark/experiments.py

```diff
@@ -43,6 +43,57 @@
 # fmt: off
 
 
+class _Constraint:
+
+    def _function(self, data: np.ndarray) -> float:
+        raise NotImplementedError()
+
+    def __call__(self, data: np.ndarray) -> float:
+        if not isinstance(data, np.ndarray):
+            raise ValueError(f"Unexpected inputs as np.ndarray, got {data}")
+        return self._function(data)
+
+
+class _positive_sum_float(_Constraint):
+    def _function(self, data: np.ndarray) -> float:
+        return float(np.sum(data))
```

Comment:
This is not what I expressed, since this is just as long. Please do this instead, which factorizes a lot of code:

```suggestion
class _Constraint:

    def __init__(self, name: str, as_bool: bool) -> None:
         self.name = name
         self.as_bool = as_bool

    def __call__(self, data: np.ndarray) -> tp.Union[bool, float]:
        if not isinstance(data, np.ndarray):
            raise ValueError(f"Unexpected inputs as np.ndarray, got {data}")
        if self.name == "positive_sum_float":
            value = float(np.sum(data))
        elif self.name ==...:
             ...
         else:
             raise NotImplentedError(f"Unknown function {self.name}")
         return value > 0 if self.as_bool else value
```

Also I would consider renaming the function, positive_sum_float is not positive, so probably just `sum`, and since summing would get a float, no need for `_float`.  And the `as_bool` (or `positive_indicator`?) flag can provides the equivalent in boolean form without more code.